pkg/providers/property/data_akamai_search.go

```diff
@@ -0,0 +1,115 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/akamai/AkamaiOPEN-edgegrid-golang/v2/pkg/papi"
+	"github.com/akamai/terraform-provider-akamai/v2/pkg/akamai"
+	"github.com/akamai/terraform-provider-akamai/v2/pkg/tools"
+)
+
+func dataSourceSearch() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceSearchRead,
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Key must have one of three values: \"edgeHostname\", \"hostname\" or \"propertyName\"",
+				ValidateDiagFunc: tools.IsNotBlank,
+			},
+			"value": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Value of Search",
+				ValidateDiagFunc: tools.IsNotBlank,
+			},
+			"properties": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of properties",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"account_id":        {Type: schema.TypeString, Computed: true},
+						"asset_id":          {Type: schema.TypeString, Computed: true},
+						"contract_id":       {Type: schema.TypeString, Computed: true},
+						"group_id":          {Type: schema.TypeString, Computed: true},
+						"property_id":       {Type: schema.TypeString, Computed: true},
+						"property_version":  {Type: schema.TypeInt, Computed: true},
+						"property_name":     {Type: schema.TypeString, Computed: true},
+						"edge_hostname":     {Type: schema.TypeString, Computed: true},
+						"hostname":          {Type: schema.TypeString, Computed: true},
+						"production_status": {Type: schema.TypeString, Computed: true},
+						"staging_status":    {Type: schema.TypeString, Computed: true},
+						"updated_by_user":   {Type: schema.TypeString, Computed: true},
+						"updated_date":      {Type: schema.TypeString, Computed: true},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceSearchRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	meta := akamai.Meta(m)
+	client := inst.Client(meta)
+
+	log := meta.Log("PAPI", "dataSourceSearchRead")
+
+	log.Debug("Read Search")
+
+	var key, value string
+	var err error
+
+	if key, err = tools.GetStringValue("key", d); err != nil {
+		return diag.FromErr(err)
+	}
+	if value, err = tools.GetStringValue("value", d); err != nil {
+		return diag.FromErr(err)
+	}
+
+	request := papi.SearchRequest{
+		Key:   key,
+		Value: value,
+	}
+
+	search, err := client.SearchProperties(ctx, request)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("could not load properties: %w", err))
+	}
+
+	d.SetId(request.Key + request.Value)
+
+	if err := d.Set("properties", sliceResponseSearch(search)); err != nil {
+		return diag.Errorf("error setting properties: %s", err)
+	}
+
+	return nil
+}
+
+func sliceResponseSearch(propertiesResponse *papi.SearchResponse) []map[string]interface{} {
```

Comment:
I am not sure, I have not checked this code on my computer. But is it really necessary this function? What if you just do a simple `d.Set("properties", search.Versions.Items)` in line 87?

Reply:
OK, I see that it would not work. Now this detail seems fine to me.